packages/js/src/feeds/Feeds.ts

```diff
@@ -1,9 +1,34 @@
-import { NovuEventEmitter } from '../event-emitter';
+import type { PaginatedResponse } from '../types';
+import { BaseModule } from '../utils/base-module';
+import { Notification } from './notification';
 
-export class Feeds {
-  #emitter: NovuEventEmitter;
+interface FetchFeedOptions {
+  page?: number;
+  feedIdentifier?: string | string[];
+  seen?: boolean;
+  read?: boolean;
+  limit?: number;
+  payload?: Record<string, unknown>;
+}
+
+export class Feeds extends BaseModule {
+  async fetch({ page = 1, ...restOptions }: FetchFeedOptions): Promise<PaginatedResponse<Notification>> {
+    return this.callWithSession(async () => {
```

Comment:
the wrapper function that awaits for the session initialization before calling the code